analyzere_extras/visualizations.py

```diff
@@ -0,0 +1,341 @@
+import hashlib
+
+from analyzere import LayerView
+from graphviz import Digraph
+try:
+    from IPython.display import FileLink
+    _in_notebook = True
+except ImportError:
+    _in_notebook = False
+
+from sys import float_info
+
+
+def _counter():
+    """Simple counter for generating unique indexes in a graph"""
+    i = 0
+    while True:
+        i += 1
+        yield i
+
+
+def _format_description(description):
+    """Clean the description of a node for diaply in Graphviz"""
+    return description.replace('\'', '').encode('unicode_escape').decode()
+
+
+def _format_DateField(df):
+    """Format a Date field, chopping off the time portion"""
+    return df.date().isoformat()
+
+
+def _format_MoneyField(mf):
+    """Format a MoneyField to '<unlimited|value> CCY' """
+    formatted = 'n/a'
+    if mf.value == float_info.max:
+        formatted = 'unlimited'
+    else:
+        formatted = '{:,.0f} {}'.format(float(mf.value), mf.currency)
+    return formatted
+
+
+def _format_layer_terms(layer):
+    """Get the terms for the given layer for display in Graphviz"""
+    terms = ''
+    if hasattr(layer, 'inception_date') or hasattr(layer, 'expiry_date'):
+        terms += '\ncoverage=['
+        if hasattr(layer, 'inception_date'):
+            terms += '{}'.format(
+                _format_DateField(layer.inception_date))
+        else:
+            terms += '-inf'
+
+        if hasattr(layer, 'expiry_date'):
```

Comment:
it could kill the extra if/else nesting by doing so, since you could (at the end) check `if date_list: terms += coverage + joined:`